azure/services/privateendpoints/spec.go

```diff
@@ -82,55 +84,182 @@ func (s *PrivateEndpointSpec) Parameters(ctx context.Context, existing interface
+		}
+
+		normalizedExistingPE := network.PrivateEndpoint{
+			Name:     existingPE.Name,
+			Location: existingPE.Location,
+			PrivateEndpointProperties: &network.PrivateEndpointProperties{
+				Subnet: &network.Subnet{
+					ID: existingPE.Subnet.ID,
+					SubnetPropertiesFormat: &network.SubnetPropertiesFormat{
+						PrivateEndpointNetworkPolicies:    existingPE.Subnet.PrivateEndpointNetworkPolicies,
+						PrivateLinkServiceNetworkPolicies: existingPE.Subnet.PrivateLinkServiceNetworkPolicies,
+					},
+				},
+				ApplicationSecurityGroups:  existingPE.ApplicationSecurityGroups,
+				IPConfigurations:           existingPE.IPConfigurations,
+				CustomNetworkInterfaceName: existingPE.CustomNetworkInterfaceName,
+			},
+			Tags: existingPE.Tags,
+		}
+
+		existingPrivateLinkServiceConnections := make([]network.PrivateLinkServiceConnection, 0, len(*existingPE.PrivateLinkServiceConnections))
+		for _, privateLinkServiceConnection := range *existingPE.PrivateLinkServiceConnections {
+			existingPrivateLinkServiceConnections = append(existingPrivateLinkServiceConnections, network.PrivateLinkServiceConnection{
+				Name: privateLinkServiceConnection.Name,
+				PrivateLinkServiceConnectionProperties: &network.PrivateLinkServiceConnectionProperties{
+					PrivateLinkServiceID: privateLinkServiceConnection.PrivateLinkServiceID,
+					RequestMessage:       privateLinkServiceConnection.RequestMessage,
+					GroupIds:             privateLinkServiceConnection.GroupIds,
+				},
+			})
+		}
+		normalizedExistingPE.PrivateEndpointProperties.PrivateLinkServiceConnections = &existingPrivateLinkServiceConnections
+
+		existingManualPrivateLinkServiceConnections := make([]network.PrivateLinkServiceConnection, 0, len(*existingPE.ManualPrivateLinkServiceConnections))
+		for _, manualPrivateLinkServiceConnection := range *existingPE.ManualPrivateLinkServiceConnections {
+			existingManualPrivateLinkServiceConnections = append(existingManualPrivateLinkServiceConnections, network.PrivateLinkServiceConnection{
+				Name: manualPrivateLinkServiceConnection.Name,
+				PrivateLinkServiceConnectionProperties: &network.PrivateLinkServiceConnectionProperties{
+					PrivateLinkServiceID: manualPrivateLinkServiceConnection.PrivateLinkServiceID,
+					RequestMessage:       manualPrivateLinkServiceConnection.RequestMessage,
+					GroupIds:             manualPrivateLinkServiceConnection.GroupIds,
+				},
+			})
+		}
+		normalizedExistingPE.PrivateEndpointProperties.ManualPrivateLinkServiceConnections = &existingManualPrivateLinkServiceConnections
+
+		// Sort all slices in order to get the same order of elements for both new and existing
+		// Sort ManualPrivateLinkServiceConnections
+		if normalizedExistingPE.ManualPrivateLinkServiceConnections != nil {
```

Comment:
It looks like calling `sort.SliceStable` on a nil slice doesn't panic if that's the reason for this condition: https://go.dev/play/p/pm_QvDXlnFJ

Reply:
Yup, I know, it I added it in order to avoid `panic: runtime error: invalid memory address or nil pointer dereference
` as I'm referencing it as a pointer to slice in SliceStable